File: backend/src/main/java/io/ntf/api/infrastructure/jwt/authentication/JwtAuthentication.java
package io.ntf.api.infrastructure.jwt.authentication;

import com.auth0.jwt.JWTVerifier;
import com.auth0.jwt.exceptions.JWTVerificationException;
import org.springframework.security.core.Authentication;

public interface JwtAuthentication {

    String getToken();

    String getKeyId();

    Authentication verify(JWTVerifier verifier) throws JWTVerificationException;
}

File: backend/src/main/kotlin/io/ntf/api/infrastructure/jwt/authentication/JwtAuthentication.kt
package io.ntf.api.infrastructure.jwt.authentication

import com.auth0.jwt.JWTVerifier
import com.auth0.jwt.exceptions.JWTVerificationException
import org.springframework.security.core.Authentication

interface JwtAuthentication {

    val token: String

    val keyId: String

    @Throws(JWTVerificationException::class)
    fun verify(verifier: JWTVerifier): Authentication
}
